### DISCUSSION TITLE
Update field value before save

### DISCUSSION THREAD

**Discussion by mikee1209** [Budibase/budibase#12540] [Help] [answered]
Hi, I want to update field value before saving it to table. I have added action to update said value, then to validate form and save row to a button. When I click it for a split second I can see the updated value but when I check Data view the value is missing. What can be the reason for that?

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @mikee1209 

This may be due to the order of operations, presumably your actions look like this:

1. You have a form that is currently in an invalid state
2. You click a button that triggers this action sequence:

- Update Field Value (presumably this would make your form valid?)
- Validate Form (this might be failing because it is still using the form state at the point the actions were triggered, i.e. invalid)
- Save Row (is not called is the validation fails)

Do I have this right? Screenshots, recordings and minimum reproduction app exports are always useful!

**Reply by melohagan** [reply]
Or if the validation is passing, then it could be the case that the **Update Field Value** doesn't update the form value until after the actions are complete.

In that case, instead of using the **Update Field Value** action, you would simply want to override a column in your **Save Row** action to include a binding value:

![Screenshot 2023-12-11 at 13 18 34](https://github.com/Budibase/budibase/assets/101575380/fa51bb8e-3feb-4f16-ad34-e4d12f2c354e)

**Reply by matteolavaggi** [reply]
Hi, thanks for the reply but I can't find a reference on how this sequence of actions should be done.

Based on my personal sense and budibase doc the step must be:

1) update field value
2) save row

why the update field value are not recorded in the save row action if this is AFTER the update field value?

If i manual add a button to update field value and next only use the save button to perform save row this work.

Seem a budibase bug this one.

This sequence does not store the update value on database

![image](https://github.com/Budibase/budibase/assets/24798703/f973c3d1-9eae-4e81-95c7-84cf10e2f8ad)
